lib/ash_json_api/controllers/helpers.ex

```diff
@@ -425,13 +425,13 @@ defmodule AshJsonApi.Controllers.Helpers do
 
           query =
             query
-            |> Ash.Query.load(fields_to_load)
-            |> Ash.Query.set_context(request.context)
             |> Ash.Query.for_read(
               action,
               Map.merge(request.arguments, params),
               Keyword.put(Request.opts(request), :page, false)
             )
+            |> Ash.Query.load(fields_to_load)
+            |> Ash.Query.set_context(request.context)
```

Comment:
I think these should remain ahead of the `for_read` call.

Reply:
@zachdaniel see my comment [here](https://github.com/ash-project/ash_json_api/pull/187#discussion_r1651704332) do you have any idea why those args would get smashed when the `load` call comes before `for_read`?